engine/execution/ingestion/engine.go

```diff
@@ -1094,6 +1094,26 @@ func (e *Engine) saveExecutionResults(
 		startState = endState
 	}
 
+	// before persisting state commitment, check whether this block has been sealed already.
+	// if the block has been sealed, it means some other execution node has executed this block
+	// already, and its result has already been accepted by the consensus nodes. In that
+	// case, we need to double check whether our execution result is consistent with the sealed
+	// result. If inconsistent, it means there is a fork in the execution results, which
+	// would be abnormal. we'd better stop the node in this case.
+	commit, err := e.state.AtBlockID(blockID).Commit()
+	if err == nil {
```

Comment:
I think we should distingush here between the expected error `storage.ErrNotFound` and an unexpected error, which should terminate the node.

Reply:
Note, the check is `err == nil` not `err != nil`.

if there is an error getting sealed commit, I would consider there isn't enough data for me to check and make decision here, so I simply ignored the commit, and continue with the normal operation.

Reply:
👍